source/components/molecules/ImageUploader/ImageUploader.tsx

```diff
@@ -118,13 +119,25 @@ const ImageUploader: React.FC<Props> = ({
   colorSchema,
   maxImages,
   id,
+  imageNamePrefix,
+  ...rest
 }) => {
   const [choiceModalVisible, toggleModal] = useModal();
 
+  const renameImagesWithPrefix = (images: Image[], prefix: string) =>
+    images.map((image, idx) => ({
+      ...image,
+      filename: `${prefix}_${idx}`,
+    }));
+
```

Comment:
It's a better practise to not loop in a function, but instead loop outside of it and make the function do one thing only like:

```js
const renameImageWithPrefix = (image: Image, prefix: string) => (
  {
   ...image,
   fileName: `${prefix}_${idx}`,
  }
);

if (imageNamePrefix) {
  updatedImages = updatedImages.map((image) => 
    renameImageWithPrefix(updatedImage, imageNamePrefix));
}
```